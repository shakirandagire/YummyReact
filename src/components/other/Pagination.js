import React from 'react';

/**
 * Component for pagination
 */

const Pagination = (props) => {
  const { perPage, total, handlePage } = props;
  const pages = [];
  let pageTotal = Math.floor(total / perPage);

  if ((total % perPage) > 0) {
    pageTotal += 1;
  }
  for (let page = 1; page <= pageTotal; page++) {
    pages.push(<p key={page} >
      <button className="black" to="#" onClick={(event) => { handlePage(event, perPage, page); }}>
        {page}
      </button>
    </p>);
  }
  return (
    <div className="paginate">
        Pages: {pages}
    </div>
  );
};
export default Pagination;
