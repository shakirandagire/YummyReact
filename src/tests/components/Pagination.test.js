import React from 'react';
import Pagination  from '../../components/pagination';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

const component = shallow(< Pagination/>);

describe('<Pagination/>',() => {
  it('should render properly', () =>{
    expect(component.length).toBe(1)
  })
  it('should render pagination',() =>{
    expect(shallowToJson(component)).toMatchSnapshot();
})
})

