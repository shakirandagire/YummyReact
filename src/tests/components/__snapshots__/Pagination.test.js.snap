// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> should render pagination 1`] = `
<div
  className="paginate"
>
  Pages: 
</div>
`;
